File: module_base/src/main/java/com/app/base/router/AppRouter.java
package com.app.base.router;

import android.net.Uri;

import com.alibaba.android.arouter.facade.template.IProvider;

/**
 * @author Ztiany
 * Email: 1169654504@qq.com
 * Date : 2017-11-04 13:44
 */
public interface AppRouter {

    IPostcard build(String path);

    IPostcard build(Uri path);

    void inject(Object object);

    <T extends IProvider> T navigation(Class<T> iProviderClass);

}

File: common_sdk/src/main/java/com/app/base/router/AppRouter.kt
package com.app.base.router

import com.android.common.router.Navigator

/**
 * @author Ztiany
 * Date : 2017-11-04 13:44
 */
interface AppRouter {

    fun <T : Navigator> get(clazz: Class<T>): T?

    fun <T : Navigator> require(clazz: Class<T>): T

}